theano/tensor/basic.py

```diff
@@ -3833,7 +3848,7 @@ def stack(*tensors):
         tensors = list(map(as_tensor_variable, tensors))
         dtype = scal.upcast(*[i.dtype for i in tensors])
         return theano.tensor.opt.MakeVector(dtype)(*tensors)
-    return join(0, *[shape_padleft(t, 1) for t in tensors])
+    return join(axis, *[shape_padleft(t, 1) for t in tensors])
```

Comment:
Hum... shape_padleft is adding an empty dimension on axis 0 (while shifting others 1 dimension higher). It works for axis=0 but if I understand correctly the added empty dimension should be at `axis`, no 0. You would need to define a third function shape_padaxis(t, axis, n_ones=1).